src/ripple/app/tx/impl/CashCheck.cpp

```diff
@@ -376,10 +508,14 @@ CashCheck::doApply()
                         << "flow did not produce DeliverMin.";
                     return tecPATH_PARTIAL;
                 }
-                if (doFix1623)
+                if (doFix1623 && !checkCashMakesTrustLine)
                     // Set the delivered_amount metadata.
                     ctx_.deliver(result.actualAmountOut);
             }
+            // Set the delivered amount metadata in all cases, not just
```

Comment:
Prior to `checkCashMakesTrustLine`, why was the delivered amount metadata set only if DeliverMin was used?

Reply:
Hi @intelliot, I believe that was a bug.  As part of this pull request I added some unit tests.  I expected that in all cases the result of a successful `CheckCash` transactions would include a `delivered_amount` field.  I found that `delivered_amount` was not always set.  I surmised `delivered_amount` was not set because `DeliveredAmount` was not always set due to the conditional above.

When I changed the code so setting `DeliveredAmount` was only conditional on the amendment then the unit tests returned a `delivery_amount` field in all of the tested cases.